skimage/exposure/hdr.py

```diff
@@ -0,0 +1,311 @@
+import numpy as np
+
+
+def get_radiance(images, exposure, radiance_map):
+    """
+    Return the radiance image for a series of images based upon a camera
+    response function.
+
+    Parameters
+    ----------
+    images: list
+        List of images in the for of numpy arrays. Either mono or color
+        (RGB, MxNx3).
+    exposure : numpy 1D array
+        Array of exposure times in seconds.
+    radiance_map : numpy array
+        Array mapping the counts to radiance
+
+    Returns
+    -------
+    hdr : numpy array
+        Resulting image with radiance values.
+    """
+
+    den = np.ones(images[0].shape)
+    num = np.zeros(images[0].shape)
+    wf = np.vectorize(_weight_func)
+    for idx, im in enumerate(images):
+        gij = im.copy()
+        # For colour
+        if im.ndim == 3:
+            for ii in range(im.shape[2]):
+                gij[:, :, ii] = radiance_map[im[:, :, ii] + 1, ii]
+        else:
+            gij[:, :] = radiance_map[im[:, :, ii] + 1]
+        gij = gij - np.log(exposure[idx])
+
+        wij = wf(gij)
+        num = num + wij * gij
+        den = den + wij
+
+    return num / den
+
+
+def make_hdr(images, exposure, radiance_map, depth=16):
+    """
+    Compute the HDR image from a series of images with a given radiance
+    mapping.
+
+    Parameters
+    ----------
+    images: list or ImageCollection
+        List of images as numpy arrays. Either grayscale or color
+        (RGB).
+        Can also be an ImageCollection
+    exposure : numpy 1D array
+        Array of exposure times in seconds.
+        Images some times have these in the exif information.
+    radiance_map : numpy array
+        Array (idx) mapping counts to radiance value, if input is RGB this must
+        be Nx3. See `get_crf`.
+    depth : int, optional
+        Pixel depth.
+
+    Returns
+    -------
+    hdr : numpy array
+        The HDR image, either grayscale or RGB depending on input, in ln(E).
+
+    References
+    ----------
+    .. [1]  Debevec, P. E., & Malik, J. (1997). SIGGRAPH 97 Conf. Proc.,
+            August, 3-8. DOI:10.1145/258734.258884
+    """
+    B = np.log(np.array(exposure))
+
+    if images[0].ndim == 3:
+        sx, sy, sc = np.shape(images[0])
+        hdr = np.zeros([sx, sy, sc], dtype=np.float)
+        gray = False
+    else:
+        sx, sy = np.shape(images[0])
+        hdr = np.zeros([sx, sy], dtype=np.float)
+        gray = True
+
+    images = np.asarray(images, dtype=np.uint64)
```

Comment:
This conversion to `np.uint64` may cause troubles (if `images` dtype is float for example)... `skimage.utils.dtype.img_as_uint` can help here.

Reply:
I agree it needs to support float images, though `skimage.utils.dtype.img_as_uint` only returns a uint16, which imposes a limit on the input images.

Reply:
I assume that this conversion is necessary for using `images` values as indices for `radiance_map` [here](https://github.com/scikit-image/scikit-image/pull/1539/files/af72226d749e6fc8cba7e9f4a8b18ce0e529d7d8#diff-2b9cb3ad83fadc5503d69c41dc02d4c235c90f3932c046e8025f2c4f50a8469cR97) which is of size 2**depth*channel_count according to `get_crf` docstring.
It needs a mapping between `images` values and `radiance_map` coordinates range...

Reply:
Yes, the conversion is needed for the use of it as indices [0, 2**depth - 1]. 
So the mapping could be to `uint`_depth_ though that would only be valid if we are working with 8, 16, 32, 64... Therefore 64 has been chosen as it will never hit a limit (I have seen some publications on 19-bit CMOS chips). 
In the latest commit, I added a todo for supporting float images. I guess I could implement a conversion in this code that scales them to `depth`.

Reply:
I tried to find a way of doing this, but I have not been able to find one, so I have added a check that raises an error if a float image is inputted

Reply:
This is a fairly significant limitation; we have to think about how to work around it.